src/VueGapi/GoogleAuthService.js

```diff
@@ -254,20 +246,11 @@ export default class GoogleAuthService {
    *   }
    * </script>
    */
-  logout(onResolve, onReject) {
-    if (!this.authInstance) throw new Error('gapi not initialized')
-    return new Promise((res, rej) => {
-      this.authInstance.signOut().then(
-        () => {
-          this._clearStorage()
-          this.authenticated = false
-          res()
-        },
-        (error) => {
-          rej(error)
-        }
-      )
-    }).then(...thenArgsFromCallbacks(onResolve, onReject))
+  logout() {
+    return this.clientProvider
+      .getAuthInstance()
+      .then((authInstance) => authInstance.signOut())
+      .then(() => this._clearStorage())
```

Comment:
I prefer this approach as it's compact and clear. Rather than having `onResolve` and `onReject` methods as callbacks, letting the user use async-await on `logout()` or treat logout as a promise is a better approach.